File: esque-core/src/main/java/org/loesak/esque/core/elasticsearch/documents/MigrationLock.java
package org.loesak.esque.core.elasticsearch.documents;

import com.fasterxml.jackson.annotation.JsonTypeInfo;
import com.fasterxml.jackson.annotation.JsonTypeName;

import java.time.Instant;
import java.util.Objects;

@JsonTypeInfo(include = JsonTypeInfo.As.WRAPPER_OBJECT, use = JsonTypeInfo.Id.NAME)
@JsonTypeName(value = "lock")
public record MigrationLock(Instant date) {

    public MigrationLock {
        Objects.requireNonNull(date);
    }

}

File: esque-core/src/main/kotlin/org/loesak/esque/core/elasticsearch/documents/MigrationLock.kt
package org.loesak.esque.core.elasticsearch.documents

import com.fasterxml.jackson.annotation.JsonTypeInfo
import com.fasterxml.jackson.annotation.JsonTypeName
import java.time.Instant

@JsonTypeInfo(include = JsonTypeInfo.As.WRAPPER_OBJECT, use = JsonTypeInfo.Id.NAME)
@JsonTypeName(value = "lock")
data class MigrationLock(@get:JvmName("date") val date: Instant)
